torch/csrc/jit/codegen/cuda/arith.cpp

```diff
@@ -479,7 +479,7 @@ TORCH_CUDA_API TensorView* broadcast(
       out_domain.push_back(new IterDomain(
           new Int(0), new Int(1), ParallelType::Serial, false, false, true));
     } else {
-      out_domain.push_back(inp->axis(iinp));
+      out_domain.push_back(inp->domain()->noReductions()[iinp]);
```

Comment:
Can you add a short comment explaining why noReductions() is used?

Reply:
added in new pr